### DISCUSSION TITLE
Does a hashgrid query only iterate over values within radius?

### DISCUSSION THREAD

**Discussion by cadop** [NVIDIA/warp#62] [Q&A] [answered]
In the example https://nvidia.github.io/warp/_build/html/modules/runtime.html#hash-grids

The code creates a query `query = wp.hash_grid_query(grid, p, radius)` and then within the while loop of that query, checks `if (dist <= radius):` .   

@mmacklin  Is this redundant, or is the radius in the hash grid query not really creating a query based on that radius?

The doc: https://nvidia.github.io/warp/_build/html/modules/functions.html?highlight=hash_grid_query#warp.hash_grid_query says ` This query can be used to iterate over all neighboring points within a fixed radius from the query point. `.  So it seems redundant, but a little confusing as an example then.

**Comment by mmacklin** [comment, marked as answer]
Hi @cadop, the HashGrid will give back all points in *cells* that fall inside the query radius. When there are hash conflicts it means that some points outside of query radius will be returned, and you should check for this in your kernel as well.

The reason the query doesn't do the check itself is that usually the first thing kernels do is to compute the distance themselves, so it would be redundant (like you mentioned).

I will make this more clear in the docs.

Thanks,
Miles
